Update layout of multiplayer room screens

smoogipoo opened the pull request:
Preview: https://drive.google.com/file/d/1mxR4xUaWqsaqFYdqE0ufkTHFpqUFozJk/view?usp=sharing

Apologies for the large PR. I can split some of this out upon request, like the `DrawableRoom` refactor.

In total:
- Previously both `PlaylistsRoomSubScreen` and `MultiplayerMatchSubScreen` were left to add _everything_ including backgrounds. `RoomSubScreen` now has a bit more of a complete implementation for the screen's base structure, and exposes `CreateMainContent()` / `CreateFooter()` / `CreateRoomSettingsOverlay()` for truly custom components of the screen.
- The screens underwent some re-layout according to the new designs.
- Individual component designs have not been adjusted yet.
- The lounge-specific parts of `DrawableRoom` (selection, joining room, filtering, hover sounds) were split into `DrawableLoungeRoom`.

smoogipoo committed:
Initial refactor of RoomSubScreen

smoogipoo committed:
Adjust sizings and paddings

smoogipoo committed:
Rename MatchSettingsOverlay and related classes

Because "match" is a multiplayer-only concept.

smoogipoo committed:
Actually show the room in the sub screen

smoogipoo committed:
Move RoomSettingsOverlay to RoomSubScreen

smoogipoo committed:
Make SettingsOverlay protected

smoogipoo committed:
Remove match header

smoogipoo committed:
Fix settings overlay not being initially visible

smoogipoo committed:
Add edit button to room panel

smoogipoo committed:
Simplify by reusing same room panel

smoogipoo committed:
Merge branch 'master' into match-subscreen-redesign

smoogipoo committed:
Remove extra corner radius on DrawableRoom

smoogipoo committed:
Resolve room settings layout issues

smoogipoo committed:
Move paddings up one level

smoogipoo committed:
Fix initial RoomId state not being handled correctly

smoogipoo committed:
Initial integration of playlists with the new structure

smoogipoo committed:
Remove participants from playlists screen

smoogipoo committed:
Disable edit button in playlists

smoogipoo committed:
Revert unintentional change

smoogipoo committed:
Adjust paddings/spacings

smoogipoo committed:
Cleanup around footers

smoogipoo committed:
Split lounge-specific implementation from DrawableRoom

smoogipoo committed:
Cleanup unnecessary implementation

smoogipoo committed:
Hide border by default

frenzibyte commented:
One thing I've noticed while I was checking this:

https://user-images.githubusercontent.com/22781491/129907212-9abae845-2349-4f67-9f9c-6ffab695467b.mp4

Also the drawable room doesn't seem to be updated when selecting a different beatmap, still displays the old beatmap on its cover.

peppy commented:
> Also the drawable room doesn't seem to be updated when selecting a different beatmap, still displays the old beatmap on its cover.

Will wait for this to be resolved before jumping into this one.

smoogipoo committed:
Fix background not updated in room

smoogipoo committed:
Remove realtime room category, fix end-date showing for realtime

peppy committed:
Play settings hide animation if visible when exiting `RoomSubScreen`

peppy committed:
Remove horizontal transitions for now

They look bad with the current toolbar implementation. We can probably
add them back once the toolbar is moved to the lounge.

peppy commented on osu.Game/Screens/OnlinePlay/Lounge/Components/DrawableRoom.cs:
```diff
@@ -359,63 +240,40 @@ protected override void LoadComplete()
+        private int numberOfAvatars = 7;
 
-        protected override bool OnClick(ClickEvent e)
+        public int NumberOfAvatars
```
Not going to dwell on this, but we usually have public properties at the top right?

smoogipoo replied:
These days I put business-logic properties below the ctor, because when you have too many of them both the ctor and `load()` get pushed down and generally I'm more interested in looking at those if I'm working with a class.

My typical structure is something like:
```
1. Public consts
2. Private consts
3. Public auto/inline properties.
4. Public readonly fields.
5. Public fields.
6. Private resolved dependencies.
7. Private readonly fields.
8. Private fields
9. Ctor
10. load()
11. LoadComplete()
12. Public complex properties.
13. Whatever else (no real structure here atm, typically public methods at the top and privates at the bottom, but this doesn't always make sense).
14. (At the end of the file) Abstract methods
```

peppy committed:
Add padding between beatmap panel and edit button

peppy committed:
Move chat to right column in multiplayer match screen

peppy committed:
Start free mod selection area hidden on screen display

peppy committed:
Adjust spacing and padding of footer buttons

peppy commented:
Made some further changes (moving chat, improving transitions and paddings). Looks good otherwise 👍 .
